keel-plugin/src/main/kotlin/com/netflix/spinnaker/config/AnnealingConfiguration.kt

```diff
@@ -0,0 +1,25 @@
+package com.netflix.spinnaker.config
+
+import com.netflix.spinnaker.keel.annealing.ResourceActuator
+import com.netflix.spinnaker.keel.annealing.ResourceCheckQueue
+import com.netflix.spinnaker.keel.annealing.spring.ApplicationEventResourceCheckListener
+import com.netflix.spinnaker.keel.annealing.spring.ApplicationEventResourceCheckQueue
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class AnnealingConfiguration {
+
+  @Bean
+  @ConditionalOnMissingBean(ResourceCheckQueue::class)
+  fun resourceCheckQueue(publisher: ApplicationEventPublisher) =
+    ApplicationEventResourceCheckQueue(publisher)
+
+  @Bean
+  @ConditionalOnBean(ApplicationEventResourceCheckQueue::class)
```

Comment:
Only register the listener if the `ResourceCheckQueue` is the Spring one. I tried to put the listener method in the queue class but Spring doesn't like it because the event handler method has to be on the interface.